Add a check for zero-byte files.

desrosj opened the pull request:
Trac ticket: https://core.trac.wordpress.org/ticket/59527

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

desrosj committed:
Add a check for zero-byte files.

desrosj committed:
Add Twenty Twenty-Four to the theme test workflow.

desrosj committed:
Only look in the theme being tested.

costdev commented on .github/workflows/test-and-zip-default-themes.yml:
```diff
@@ -49,6 +49,51 @@ concurrency:
+          show-progress: ${{ runner.debug == '1' && 'true' || 'false' }}
+
+      - name: Find zero-byte files
+        run: find src/wp-content/themes/${{ matrix.theme }} -size 0
```
Something like this may work:

```yaml
if [[ "" != $(find src/wp-content/themes/${{ matrix.theme }} -size 0) ]]; then exit 1; fi
```
Not sure if that translates to the workflow properly, but you get the gist :slightly_smiling_face:

desrosj replied:
Ah! I didn't see this before making 7f5fb21dd15583893ccad63a09cae4a1890ffaf2. What do you think about that approach? I think it reads a lot easier. But not opposed to something like above either.

costdev replied:
The other approach works but I'm not sure that it'll be clear that the diff check is part of, and required for, the empty file checks.

This also works in bash and may work here:
```yaml
- name: Make sure there are no empty files.
- run: [[ ! $(find src/wp-content/themes/${{ matrix.theme }} -empty) ]]
```
Do you think that's readable enough or not quite?

desrosj replied:
Hmm, I've added this, but it doesn't seem to be flagging the empty images (I haven't synced my PR branch since that was fixed in `trunk`).

costdev replied:
Hmm, it should be producing a 1 exit code. Strange. Unless a readable oneliner comes to mind, the diff check makes most sense

desrosj replied:
Strange! I just tried it again and it did work. 🤷‍♂️

desrosj committed:
Ensure empty files are correctly flagged.

`find` always returns a successful status code, unless there is a problem traversing directories. Instead, let's delete empty files and then check that there are no changes in GIT.

desrosj committed:
Improve inline docs.

desrosj committed:
Try one liner.

desrosj committed:
Try one liner take 2.

desrosj committed:
Another attempt to find a one-liner.

desrosj committed:
Go back go using `-delete` and `git diff --exit-code`.

desrosj committed:
Figuring out why files are not deleting.

desrosj committed:
More testing.

desrosj committed:
Change to the original command.

desrosj committed:
Merge branch 'trunk' of https://github.com/WordPress/wordpress-develop into add/0-byte-file-check

desrosj committed:
Introduce 0 byte files.

desrosj committed:
Try one liner one more time.

desrosj commented:
Fixed in https://core.trac.wordpress.org/changeset/56792.
